examples/fundamentals__timeout/cypress/integration/all-tests-spec.js

```diff
@@ -0,0 +1,53 @@
+/// <reference types="cypress" />
+
+import { seconds, testTimeout } from './timeout'
+
+describe('Global timeout of 3 seconds', () => {
+  // Let's try to time limit _every test_
+
+  // first attempt: using "beforeEach" hook to call testTimeout
+  // DOES NOT WORK - the hook has its own runnable context, but
+  // does not have access to the test to be executed
+  // context.only('of 3 seconds in every test', () => {
+  //   beforeEach(() => {
+  //     testTimeout(seconds(3))
+  //   })
+
+  //   it('runs for 1 second', () => {
+  //     cy.wait(1000)
+  //   })
+
+  //   it.skip('runs for 5 seconds (should fail)', () => {
+  //     cy.wait(5000)
+  //   })
+  // })
+
+  // second attempt: using 'test:before:run' event
+  // this WORKS, make sure to pass the test object
+  // from the event callback's arguments
+  context('of 3 seconds in every test', () => {
+    // IMPORTANT: while this event listener is defined inside a suite
+    // it is called _for every test_.
+    Cypress.on('test:before:run', function (attributes, test) {
+      // during this event, we get the test instance from the arguments
+      testTimeout(seconds(3), test)
+    })
```

Comment:
why have it nested in the suite? it would be more clear to have outside all suites

Reply:
because I have another suite of tests (commented) that try to do the same thing using `beforeEach` hook - and I did not want the two approaches to overlap.